extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -141,4 +152,38 @@ public void logCleanup(BuildProducer<LogCleanupFilterBuildItem> logCleanupFilter
         logCleanupFilter.produce(new LogCleanupFilterBuildItem("io.smallrye.metrics.MetricsRegistryImpl",
                 "Register metric ["));
     }
+
+    @BuildStep
+    @Record(STATIC_INIT)
+    void registerMetricsFromAnnotatedMethods(SmallRyeMetricsRecorder metrics,
+            BeanContainerBuildItem beanContainerBuildItem,
+            BeanArchiveIndexBuildItem beanArchiveIndex) {
+        JandexBeanInfoAdapter beanInfoAdapter = new JandexBeanInfoAdapter(beanArchiveIndex.getIndex());
+        JandexMemberInfoAdapter memberInfoAdapter = new JandexMemberInfoAdapter(beanArchiveIndex.getIndex());
+        for (ClassInfo clazz : beanArchiveIndex.getIndex().getKnownClasses()) {
+            // TODO: maybe leave out more cases where we can be sure that there are no metrics to be registered
+            // to speed up the startup
+            if (clazz.name().prefix().toString().startsWith("io.smallrye.metrics")) {
+                continue;
+            }
+            BeanInfo beanInfo = beanInfoAdapter.convert(clazz);
+            for (MethodInfo method : clazz.methods()) {
+                if (containsMetricsAnnotations(method.annotations()) ||
+                        containsMetricsAnnotations(clazz.classAnnotations())) {
+                    metrics.registerMetrics(beanInfo, memberInfoAdapter.convert(method));
+                }
+            }
+        }
+    }
+
+    private boolean containsMetricsAnnotations(Collection<AnnotationInstance> annotations) {
+        return annotations.stream().anyMatch(annotation -> metricsAnnotations.contains(annotation.name()));
+    }
+
+    //    @BuildStep
+    //    @Record(STATIC_INIT)
+    void registerMetricsFromProducerFieldsAndMethods(BeanArchiveIndexBuildItem beanArchiveIndex) {
+        // TODO: Is this possible at all? to register such metric we need to actually instantiate the object produced by the field/method, which we can't do in STATIC_INIT?!
```

Comment:
So is this a regression in feature compared to the existing extension?

Reply:
No, this didn't work before either.